src/components/Auth/components/RegisterForm/RegisterForm.spec.jsx

```diff
@@ -1,231 +1,91 @@
-
-    wrapper.setState({
-      password: 'ASDFASDF',
-    });
-
-    passInput.simulate('blur', event);
-    expect(wrapper.state().err.passErr).to
-      .equal('Please use at least one lowercase letter.');
-
-    wrapper.setState({
-      password: 'asdfaSdf',
-    });
-
-    passInput.simulate('blur', event);
-    expect(wrapper.state().err.passErr).to
-      .equal('Please use at least one number.');
-
-    wrapper.setState({
-      password: 'asdfaS3f',
-    });
-
-    passInput.simulate('blur', event);
-    expect(wrapper.state().err.passErr).to
-      .equal('Please use at least one special character.');
-
-    passInput.simulate('focus', event);
-    expect(wrapper.state().err.passErr).to.equal('');
-
-    wrapper.setState({
-      password: 'asdfaS3_',
-    });
-
-    passInput.simulate('blur', event);
-    expect(wrapper.state().err.passErr).to
-      .equal('');
-  });
-
-  it('Checks email onBlur, removes errors onFocus', () => {
-    const wrapper = shallow(<Register {...props} />);
-    const input = wrapper.find({ name: 'email' });
-    const event = {
-      target: {
-        name: 'email',
-      },
+    const changePasswordEvent = {
+      target: { value: 'AnotherPassword' },
     };
-
-    wrapper.setState({
-      email: 'test@test.com',
-    });
-
-    input.simulate('blur', event);
-    expect(wrapper.state().err.emailErr).to
-      .equal('Please use a TTU email address.');
-
-    input.simulate('focus', event);
-    expect(wrapper.state().err.emailErr).to.equal('');
-
-    wrapper.setState({
-      email: 'test@ttu.edu',
-    });
-
-    input.simulate('blur', event);
-    expect(wrapper.state().err.emailErr).to
-      .equal('');
+    wrapper.setState(testState);
+    wrapper.instance().handleInputChange('password')(changePasswordEvent);
+    expect(props.validate.called).to.equal(true);
   });
 
-  it('Checks password confirmation onBlur, removes errors onFocus', () => {
-    const wrapper = shallow(<Register {...props} />);
-    const input = wrapper.find({ name: 'confirmPass' });
-    const event = {
-      target: {
-        name: 'confirmPass',
-      },
+  it('should submit the form without any changes to the values', () => {
+    const formValues = {
+      email: 'student.atTech@ttu.edu',
+      password: 'ProperPassword1!',
+      confirmPass: 'ProperPassword1',
+      name: 'Miggy',
+      role: 'student',
+      preventSubmit: false,
     };
-
-    wrapper.setState({
-      password: 'test',
-      confirmPass: 'asdf',
-    });
-
-    input.simulate('blur', event);
-    expect(wrapper.state().err.confirmErr).to
-      .equal('Please enter a matching password.');
-
-    input.simulate('focus', event);
-    expect(wrapper.state().err.confirmErr).to.equal('');
-
-    wrapper.setState({
-      password: 'test',
-      confirmPass: 'test',
-    });
-
-    input.simulate('blur', event);
-    expect(wrapper.state().err.confirmErr).to
-      .equal('');
-  });
-
-  it('Displays server errors', () => {
```

Comment:
This is something i will implement in redux. I am going with what you suggested and have a action we can dispatch to show all server errors as snackbars